### PR TITLE
Add scatter_max and scatter_min

### PR DESCRIPTION
**Pull request by rezoo**
This PR adds `cupyx.scatter_min` and `cupyx.scatter_max` to `cupy.ndarray`.

- [x] Support float64
- [x] Write tests
- [x] Write document

### CODE REVIEW

**Comment by rezoo** [comment]
I think this PR can be reviewed.

File: cupyx/scatter.py [hunk 1]
```diff
@@ -54,3 +54,77 @@ def scatter_add(a, slices, value):
...
+    Example
+    -------
+    >>> import numpy
+    >>> import cupy
+    >>> a = cupy.zeros((6,), dtype=numpy.float32)
+    >>> i = cupy.array([1, 0, 1, 2])
+    >>> v = cupy.array([1., 2., 3., -1.])
+    >>> cupyx.scatter_max(a, i, v);
+    >>> a
+    array([2., 3., 0., 0., 0., 0.], dtype=float32)
```
Comment (emcastillo) [inline cupyx/scatter.py]: Probably I am mistaken but shouldn't a be ([2,3,-1, ...])?

Comment (rezoo) [reply cupyx/scatter.py]: Thank you for your suggestion.
Exactly, in ``scatter_max``, the initial value of `a` is also taken into account. I revised the documents.
I also wrote that the specification of ``scatter_max`` is almost the same as ``numpy.maximum.at``. I'd be glad if you could check the documents.

**Comment by emcastillo** [comment]
Jenkins, test this please

File: cupyx/scatter.py [hunk 2]
```diff
@@ -54,3 +54,83 @@ def scatter_add(a, slices, value):
...
+    -------
+    >>> import numpy
+    >>> import cupy
+    >>> a = cupy.zeros((6,), dtype=numpy.float32)
+    >>> i = cupy.array([1, 0, 1, 2])
+    >>> v = cupy.array([1., 2., 3., -1.])
+    >>> cupyx.scatter_min(a, i, v);
+    >>> a
+    array([0., 0., -1., 0., 0., 0.], dtype=float32)
+
```
Comment (emcastillo) [inline cupyx/scatter.py]: Docstring generation is failing here

13:46:34 File "../../cupyx/scatter.py", line ?, in default
13:46:34 Failed example:
13:46:34     a
13:46:34 Expected:
13:46:34     array([0., 0., -1., 0., 0., 0.], dtype=float32)
13:46:34 Got:
13:46:34     array([ 0.,  0., -1.,  0.,  0.,  0.], dtype=float32)
13:46:34 
13:46:35 Makefile:176: recipe for target 'doctest' failed
13:46:35 make: *** [doctest] Error 2
13:46:35 ERROR:root:Failed to run test
13:46:35 ERROR:root:Exit code: 2
13:46:35 cloning chainer/chainer master

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by rezoo** [comment]
Sorry, I fixed the wrong docstring.

**Comment by emcastillo** [comment]
jenkins, test this please

**Comment by rezoo** [comment]
The build on Jenkins failed, but its reason seems not to be related to my changes.

**Comment by emcastillo** [comment]
Exactly, lets merge it.

Thanks!

**Pull request merged**